client/verta/tests/test_model_registry/test_model_version.py

```diff
@@ -166,3 +166,18 @@ def test_archive(self, model_version):
             model_version.archive()
 
         assert "the version has already been archived" in str(excinfo.value)
+
+    def test_clear_cache(self, registered_model):
+        # Multiple log_artifacts calls, which would potentially fail without clear_cache
+        model_version = registered_model.get_or_create_version(name="my version")
+        model_version_2 = registered_model.get_version(id=model_version.id) # same version object
+
+        np = pytest.importorskip("numpy")
+        artifact = np.random.random((36, 12))
+
+        for i in range(5):
+            model_version.log_artifact("artifact_{}".format(2 * i), artifact)
+            model_version_2.log_artifact("artifact_{}".format(2 * i + 1), artifact)
+
+        model_version = registered_model.get_version(id=model_version.id) # re-retrieve the version
+        assert len(model_version._msg.artifacts) == 10
```

Comment:
I think 10 artifacts is a bit overkill, and gives the impression that this test isn't really confident in its ability to reliably catch what it's testing, if that makes sense.

Based on what you've seen, do you think we could accomplish this with, say, just 2 or 3 calls to `log_*()`? If not, this is fine for now. Let me know what you think.

Reply:
2 is a bit more hit or miss, but 4 does fail quite consistently.

Reply:
No flaky tests. What is the source of flakiness? Are you relying on a race condition? All the code should be deterministic (up to where we can).

Reply:
My guess is that due to these lines:
https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/_tracking/entity.py#L49-L55
If the time between the declaration of model_version_2 and its first log_artifact call is more than 5 seconds, then it will automatically re-retrieve the message (L53), so the test will still pass with refresh_cache instead of fetch_msg_no_cache.

Reply:
fetch_msg_no_cache will always reliably pass. refresh_cache, well if it's fast enough will fail.